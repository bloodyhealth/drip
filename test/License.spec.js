import React from 'react'
import { render, screen } from '@testing-library/react-native'
import License from '../components/settings/License'

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (str, options) => {
      return str + (options ? JSON.stringify(options) : '')
    },
  }),
}))

describe('License screen', () => {
  test('It should have a correct year', async () => {
    render(<License setLicense={() => {}} />)
    const year = new Date().getFullYear().toString()
    screen.getByText(year, { exact: false })
  })
})
