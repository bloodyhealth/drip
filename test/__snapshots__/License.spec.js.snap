// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`License screen It should match the snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#E9F2ED",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Array [
        Object {
          "backgroundColor": "#E9F2ED",
          "flexGrow": 1,
        },
        undefined,
      ]
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#555",
              "fontFamily": "Jost-Book",
              "fontSize": 34.285714285714285,
            },
            Object {
              "alignSelf": "center",
              "color": "#3A2671",
              "fontFamily": "Jost-Bold",
              "fontSize": 51.42857142857143,
              "fontWeight": "700",
              "marginHorizontal": 34.285714285714285,
              "marginVertical": 42.857142857142854,
            },
          ]
        }
      >
        title
      </Text>
      <View
        style={
          Object {
            "marginBottom": 34.285714285714285,
            "marginHorizontal": 34.285714285714285,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#555",
                "fontFamily": "Jost-Book",
                "fontSize": 34.285714285714285,
              },
              undefined,
            ]
          }
        >
          text{"currentYear":2022}
        </Text>
        <Text
          onPress={[Function]}
          style={
            Object {
              "color": "#3A2671",
              "fontFamily": "Jost-Book",
              "fontSize": 34.285714285714285,
              "textDecorationLine": "underline",
            }
          }
        >
          https://www.gnu.org/licenses/gpl-3.0.html
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
